packages/ag-charts-community/src/scene/shape/shape.ts

```diff
@@ -180,12 +212,26 @@ export abstract class Shape<D = any> extends Node<D> {
     }
 
     protected applyFill(ctx: CanvasContext) {
+        const bbox = this.getGradientFillBBox();
         ctx.fillStyle =
-            this.fillGradient?.createGradient(ctx as any, this.getBBox()) ??
+            this.fillGradient?.createGradient(ctx as any, bbox) ??
             (typeof this.fill === 'string' ? this.fill : undefined) ??
             'black';
     }
 
+    protected getGradientFillBBox() {
+        const { fill } = this;
+        if (!isGradientFill(fill)) {
+            return this.getBBox();
+        }
+
+        if (fill.bounds === 'axes' || fill.bounds === 'series') {
+            return this.gradientFillOptions.bbox ?? this.getBBox();
+        }
```

Comment:
The bounds being `axes` or `series` doesn't seem like a concern of the `Shape` type - we should extract this out to be a concern of the series implementation(s).